numpy/core/tests/test_multiarray.py

```diff
@@ -4672,6 +4672,32 @@ def test_load_object_array_fromfile(self):
         assert_raises_regex(ValueError, "Cannot read into object array",
                             np.fromfile, self.filename, dtype=object)
 
+    def test_fromfile_offset(self):
+        f = open(self.filename, 'wb')
+        self.x.tofile(f)
+        f.close()
+
+        with open(self.filename, 'r+b') as f:
+            y = np.fromfile(f, dtype=self.dtype, offset=0)
+            assert_array_equal(y, self.x.flat)
+
+        with open(self.filename, 'r+b') as f:
+            offset_items = len(self.x.flat) // 2
+            offset_bytes = self.dtype.itemsize * offset_items
+            y = np.fromfile(f, dtype=self.dtype, offset=offset_bytes)
+            assert_array_equal(y, self.x.flat[offset_items:])
```

Comment:
Would be good to check that calling `fromfile` twice with the same offset gives the same result - ie, that the `seek` calls do not stack.

Reply:
Is that the interface we desire? I would prefer one that `seek`s from `CURRENT` so you can use `fromfile` in a loop to skip over padding info between stored arrays

Reply:
The default for `file_handle.seek` is to go from the start, so perhaps that is best here as well? On the other hand, when one passes in a file handle, the current offset is used, so that argues for doing the offset from the current position. Perhaps in analogy with regular file handles, we should also have a `whence` argument?

That said, I think for a loop the best thing really is to open the file outside and seek there as well, just passing in the file handle to `fromfile` (that works fine; will start at the current offset). I feel the `offset` argument here is most useful when passing in filenames.

Reply:
Decision needed. Should `SEEK_SET` vs. `SEEK_CUR` reach the mailing list?

Reply:
At present I think it `dup`s the fd - so I don't think the type of seeking will affect things iteratively either way. Perhaps I misread that.

Either way, I'd like to see a passing test here proving which choice @fivemok's implementation makes.

Reply:
It looks like the current file position is kept track of here:

https://github.com/numpy/numpy/blob/38c8e4d27bae3fdba52bfafe1385438476df4efe/numpy/core/src/multiarray/multiarraymodule.c#L2065
https://github.com/numpy/numpy/blob/38c8e4d27bae3fdba52bfafe1385438476df4efe/numpy/core/src/multiarray/multiarraymodule.c#L2089

So I think ```SEEK_CUR``` is appropriate and subsequent calls should stack.

Reply:
I've added a case for subsequent seeks with ```SEEK_CUR```.